runner/src/runner.rs

```diff
@@ -0,0 +1,1127 @@
+use crate::{error::Error, Opts};
+use backoff::{retry, Error as BackoffError, ExponentialBackoff};
+use bytes::Bytes;
+use codec::Decode;
+use futures::{future::BoxFuture, FutureExt, StreamExt, TryFutureExt};
+use nix::{
+	sys::signal::{self, Signal},
+	unistd::Pid,
+};
+use reqwest::Url;
+use sha2::{Digest, Sha256};
+use signal_hook_tokio::SignalsInfo;
+use sp_core::{hexdisplay::AsBytesRef, H256};
+use std::{
+	convert::TryInto,
+	fmt::{Debug, Display},
+	fs::{self, OpenOptions},
+	io::{Read, Seek, SeekFrom, Write},
+	os::unix::prelude::OpenOptionsExt,
+	path::PathBuf,
+	process::{Child, Command, Stdio},
+	str::{self, FromStr},
+	time::Duration,
+};
+
+use subxt::{dynamic::Value, OnlineClient, PolkadotConfig};
+
+use async_trait::async_trait;
+
+/// Type of the client to run.
+/// Also used as the name of the downloaded executable.
+#[derive(Debug, Clone)]
+pub enum ClientType {
+	Vault
+}
+
+impl FromStr for ClientType {
+	type Err = crate::Error;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		let client_type = match s {
+			"vault" => ClientType::Vault,
+			_ => return Err(Error::ClientTypeParsingError),
+		};
+		Ok(client_type)
+	}
+}
+
+impl Display for ClientType {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		let s = match self {
+			ClientType::Vault => "vault",
+		};
+		write!(f, "{s}")
+	}
+}
+
+/// Pallet in the parachain where the client release is assumed to be stored
+pub const PARACHAIN_MODULE: &str = "ClientInfo";
+
+/// Storage item in the Pallet where the client release is assumed to be stored
+pub const CURRENT_RELEASES_STORAGE_ITEM: &str = "CurrentClientReleases";
+
+/// Parachain block time
+pub const BLOCK_TIME: Duration = Duration::from_secs(6);
+
+/// Timeout used by the retry utilities: One minute
+pub const RETRY_TIMEOUT: Duration = Duration::from_millis(60_000);
+
+/// Waiting interval used by the retry utilities: One minute
+pub const RETRY_INTERVAL: Duration = Duration::from_millis(1_000);
+
+/// Multiplier for the interval in retry utilities: Constant interval retry
+pub const RETRY_MULTIPLIER: f64 = 1.0;
+
+/// Data type assumed to be used by the parachain to store the client release.
+/// If this type is different from the on-chain one, decoding will fail.
+#[derive(Decode, Default, Eq, PartialEq, Debug, Clone)]
+pub struct ClientRelease {
+	/// A link to the Releases page of the `interbtc-clients` repo.
+	/// Example: https://github.com/interlay/interbtc-clients/releases/download/1.16.0/vault-parachain-metadata-interlay
```

Comment:
Let's adjust the comment and replace the link with this one. 
`https://downloads.pendulumchain.tech/index.html`